### PR TITLE
Add dashloader support. Again.

### PR DESCRIPTION
**Pull request by alphaqu**
Title says it all. 

Not the most clean code every but its designed to change as little of the original code as possible.

### CODE REVIEW

**Comment by PweSol** [comment]
Doesn't work. In fact, it sometimes also does this, instead of making the chest invisible as usual.

**Comment by alphaqu** [comment]
what mods are you using?

**Comment by PweSol** [comment]
I cannot replicate the issue anymore. Could have been an issue with cache. However, I'm still having another issue; the chest and the shulker closing animations being glithy when I have my mods enabled. Also, when I disable DashLoader, the issue persists. So it shouldn't be caused by it. Maybe it has to do with some rendering mod. 
(this issue doesn't happen when using only DashLoader, EBE, and fabric api)

I'm using pretty much all of the performance enhancing mods available for fabric, and a few other. So 36 mods in total. I can try figuring out on my own which mod combination is causing this issue. But if you want, I'll give you the whole list. The only bugs I ever have are always tied to either DashLoader, or EVE.

**Comment by PweSol** [comment]
And what exactly is this commit supposed to change? In game, I see no difference in behavior between it and the stable version, with and without mods.

**Comment by DragonEggBedrockBreaking** [comment]
> And what exactly is this commit supposed to change? In game, I see no difference in behavior between it and the stable version, with and without mods.

It makes EBE compatible with the DashLoader mod

File: gradle.properties [hunk 1]
```diff
@@ -6,6 +6,9 @@ loader_version=0.12.8
 
 fabric_version=0.44.0+1.18
 
+dashloader_core_version=1.2.1
+dashloader_definitions_version=3.0-rc14-1.18
```
Comment (FoundationGames) [inline gradle.properties]: These lines should go next to other dependencies

File: src/main/resources/fabric.mod.json [hunk 2]
```diff
@@ -2,7 +2,6 @@
   "schemaVersion": 1,
   "id": "enhancedblockentities",
   "version": "${version}",
-
```
Comment (FoundationGames) [inline src/main/resources/fabric.mod.json]: Wondering as to why the mod json is being reformatted

File: src/main/java/foundationgames/enhancedblockentities/compat/dashloader/DashDynamicBakedModel.java [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+        }
+
+        public DynamicModelEffects getModelEffect() {
+            return switch (this) {
+                case DEFAULT -> DynamicModelEffects.DEFAULT;
+                case CHEST -> DynamicModelEffects.CHEST;
+                case BELL -> DynamicModelEffects.BELL;
+                case SHULKER_BOX -> DynamicModelEffects.SHULKER_BOX;
+            };
+        }
```
Comment (FoundationGames) [inline src/main/java/foundationgames/enhancedblockentities/compat/dashloader/DashDynamicBakedModel.java:115]: Making these hardcoded enums for ModelSelector and ModelEffects is really not ideal as stated on the last version of this PR

**Review by FoundationGames — CHANGES_REQUESTED** [review]

**Comment by PweSol** [comment]
> > And what exactly is this commit supposed to change? In game, I see no difference in behavior between it and the stable version, with and without mods.
> 
> It makes EBE compatible with the DashLoader mod

But the stable release works with it. I'm don't understand code, so I don't really know what's going on here anyway.

`fabric_version=0.44.0+1.18`
The latest is 0.46.1+1.18
Is that an issue?

**Comment by DragonEggBedrockBreaking** [comment]
> But the stable release works with it. I'm don't understand code, so I don't really know what's going on here anyway.

The stable release is not compatible - sure, it doesn’t crash, but dashloader simply does not work - it has to recache every time and nothing happens. Also chests can look funny.

> `fabric_version=0.44.0+1.18` The latest is 0.46.1+1.18 Is that an issue?

No. The only reason it would be problematic is if the new fabric api changed code in the specific module/code that ebe is using (FAPI is a massive mod) sufficiently to break EBE. The only time this needs to be changed is either then, or if new features are added to the newest version that the mod author wants to use

**Comment by FoundationGames** [comment]
So far it looks as good as it will be, unless anyone else has any concerns this will be merged when I get the next chance (with subsequent changes)

**Comment by pluiedev** [comment]
it has been twenty. one. days.

**Comment by FoundationGames** [comment]
Please keep in mind that I work on this mod in my spare time (which has been very minimal in recent weeks) and I cannot stick to timeframes or ETAs with the development of my mods.

I also apologize if the lack of the compatibility code is resulting in a support burden in any community or support spaces.

**Pull request merged**